enhancements/installer/openstack-no-fip-install.md

```diff
@@ -0,0 +1,135 @@
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Numerous customers have requested the ability to run the OpenShift on OpenStack IPI installer without using Floating IP addresses.
+In the installer, we use floating IPs as a way to set up external connectivity to the OpenShift cluster. Depending on how the OpenStack cluster
+is set up, the use of floating IPs may be impossible.
+
+## Motivation
+
+### Goals
+
+- Extend the installer to support provider networks
+- Allow the installer to be run without setting up external connectivity
+
+### Non-Goals
+
+This enhancement does not intend to provide any guidance or features that go further than allowing customers to deploy without Floating IPs. Operations like attaching loadbalancers to the entrypoints, or any day 2 operations are beyond the scope of this enhancement. It is important to note that while we aim to support installs without external connectivity, we are not supporting installs on networks that are unable to reach the host cluster's OpenStack APIs and services.
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made:
+1. Make `externalNetwork` an optional argument in the install-config when `machinesSubnet` is set
+2. Make setting  `lbFloatingIP` when no external network is passed enforced as invalid usage
+3. Make the API and ingress VIPs dynamically assigned by default, and only assign fixed IPs if they are specified in the install config
+
+### Design Details
+
+#### Make `externalNetwork` an optional argument in the install-config when `machinesSubnet` is set
+
+Implementing this can be done by changing the validations and the defaults so that a blank external network can be passed to the installer code. We would also disable all floating IP creation in the installer when this is the case.
+
+#### Disable `lbFloatingIP` when no external network is passed
+
+In the installer, the external network is used to provision floating IPs as well as the router. The router no longer gets created when using `machinesSubnet`, so the only changes that would need to be made are changes in the installers defaults and validations that allow the a blank `lbFloatingIP` to be pased. Then we will have to make a conditional block that does not attach the floating IP when `lbFloatingIP` is blank.
+
+#### Make the API and ingress VIPs dynamically assigned by default, and only assign fixed IPs if they are specified in the install config
+
+This should be handled by adding OpenStack specific code to the baremetal-runtime-config to enable it to look up ports in a cluster. This would break the circular dependency chain the installer currently has, and allow us to allocate the VIP ports dynamically, rather than assigning fixed IPs.
```

Comment:
can we get more details on how this will be implemented. What is the circular dependency that will be broken? 
A bit more explanation perhaps.